src/main/kotlin/org/jetbrains/research/testspark/tools/llm/generation/JUnitTestsAssembler.kt

```diff
@@ -85,6 +85,30 @@ class JUnitTestsAssembler(
         log.debug(super.getContent())
     }
 
+    /**
+     * Receives a response from the HuggingFace API and processes it.
+     */
+    fun consumeHFRequest(httpRequest: HttpRequests.Request) {
+        Thread.sleep(50L)
+        val text = httpRequest.reader.readLine()
+        val generatedTestCases = removeDelimiterLines(
+            JsonParser.parseString(text).asJsonArray[0]
+                .asJsonObject["generated_text"].asString.trim(),
+        )
+        super.consume(generatedTestCases)
+        log.debug(super.getContent())
+        updateProgressBar()
+    }
+
+    /**
+     * Removes delimiter lines used for code blocks in LLMs' response
+     */
+    private fun removeDelimiterLines(text: String): String {
+        return text.lines()
+            .filter { line -> line != "```" && line != "```java" }
+            .joinToString("\n")
+    }
```

Comment:
I agree that this is a valid concern. I've moved the post-processing logic to `HuggingFaceRequestManager` in https://github.com/JetBrains-Research/TestSpark/pull/212/commits/4bce7036cbf6dac6a743aaa7332ccb5a9b13f7c3 and left a TODO to create a generic `consume(String)` later.

Reply:
> left a TODO to create a generic consume(String) later.

It already exists: `JUnitTestsAssembler` implements the `TestsAssembler` interface that has the `consume(String)` method ([here](https://github.com/JetBrains-Research/TestSpark/blob/e6adea3171f96719d5bb7853e782d9ba4e4c0ac9/src/main/kotlin/org/jetbrains/research/testspark/tools/llm/generation/JUnitTestsAssembler.kt#L46) is the `JUnitTestsAssembler`'s override). You may use it instead of `consumeHFRequest`.

Reply:
I've used `consume(String)` now (https://github.com/JetBrains-Research/TestSpark/pull/212/commits/269353925813e2542156d1031462f56ac4f7b5d9)